File: examples/reticulated_python/4.py
class A:
	def getN()->Int:
		return 2

class B:
	def foo()->String:
		return "meh"

@fields({'ref' : A, 'anyref' : Dyn})
class D:
	ref = A()
	anyref = A()
	def main(self:D) -> B:
		self.ref = A()
		self.anyref = self.ref
		return self.anyref

d = D()
d.main()

File: examples/strongscript/4.ts
class A {
	foo() : !number { return 1 }
}

class B {
	foo() : !string {return "meh"}
}

class D {
	ref : !A
	anyref : any
	main() : !B {
		this.ref = new A()
		this.anyref = this.ref
		return this.anyref
	}
}
